crates/kornia-imgproc/src/calibration/find_contours.rs

```diff
@@ -0,0 +1,287 @@
+                        if is_outer {
+                            BorderType::Outer
+                        } else {
+                            BorderType::Hole
+                        },
+                        None,
+                    ));
+                    nbd += 1;
+                }
+            }
+        }
+    }
+
+    contours
+}
+
+/// Trace a single closed border using the Suzuki–Abe method.
+///
+/// # Arguments
+///
+/// * `bin` — flattened binary image (0 or 1), row-major.
+/// * `nbrs` — 8 neighbor offsets in clockwise order: E, SE, S, SW, W, NW, N, NE.
+/// * `w, h` — image width and height.
+/// * `p0` — starting pixel coordinate (x,y), guaranteed foreground and unlabeled.
+/// * `b0` — the “backtrack” pixel (one step before `p0`), may lie out of bounds.
+///
+/// # Returns
+///
+/// A `Vec<(usize, usize)>` of all border pixels in visit order.
+fn follow_border(
+    bin: &[u8],
+    nbrs: &[(i32, i32); 8],
+    w: usize,
+    h: usize,
+    p0: (usize, usize),
+    b0: (i32, i32),
+) -> Vec<(usize, usize)> {
+    let idx = |x: usize, y: usize| y * w + x;
+    let mut contour = Vec::new();
+    let mut p = p0;
+    let mut b = b0;
+
+    loop {
+        contour.push(p);
+
+        // finding neighbour index where offset is b - p
+        let dx0 = b.0 - p.0 as i32;
+        let dy0 = b.1 - p.1 as i32;
+        let mut k0 = 0;
+        for (k, &(dx, dy)) in nbrs.iter().enumerate() {
+            if dx == dx0 && dy == dy0 {
+                k0 = k;
+                break;
+            }
+        }
+
+        // searching clockwise for next foreground neighbour from k0 + 1
+        let mut found = false;
+        let mut next_p = p;
+        let mut next_b = b;
+        for s in 1..=8 {
+            let d = (k0 + s) % 8;
+            let nx = p.0 as i32 + nbrs[d].0;
+            let ny = p.1 as i32 + nbrs[d].1;
+
+            if nx >= 0 && nx < w as i32 && ny >= 0 && ny < h as i32 {
+                let ni = idx(nx as usize, ny as usize);
+                if bin[ni] != 0 {
+                    // found the next border pixel
+                    next_p = (nx as usize, ny as usize);
+                    // backtrack = neighbor at (d-1 mod 8) relative to p
+                    let prev_d = (d + 7) % 8;
+                    let bx = p.0 as i32 + nbrs[prev_d].0;
+                    let by = p.1 as i32 + nbrs[prev_d].1;
+                    next_b = (bx, by);
+                    found = true;
+                    break;
+                }
+            }
+        }
+
+        if !found {
+            // isolated pixel (no next neighbor)
+            break;
+        }
+
+        p = next_p;
+        b = next_b;
+
+        //stop when we return to the start configuration
+        if p == p0 && b == b0 {
+            break;
+        }
+    }
+
+    contour
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use kornia_image::{Image, ImageError, ImageSize};
+
+    fn make_basic() -> Result<Image<u8, 1>, ImageError> {
```

Comment:
it's a matter of taste but generate a smaller example that defines in the code the vec to easily read/debug the image. There are other examples in the library doing it